js/src/wrappers/ai-sdk-v2.ts

```diff
@@ -0,0 +1,308 @@
+import { SpanTypeAttribute } from "@braintrust/core";
+import { startSpan } from "../logger";
+
+// Minimal interface definitions that match AI SDK v2 without importing it
+interface LanguageModelV2Middleware {
+  wrapGenerate?: (params: {
+    doGenerate: () => Promise<{
+      content: any;
+      usage?: any;
+      finishReason?: string;
+    }>;
+    params: any;
+  }) => Promise<{
+    content: any;
+    usage?: any;
+    finishReason?: string;
+  }>;
+  wrapStream?: (params: {
+    doStream: () => Promise<{
+      stream: any;
+      response?: any;
+      [key: string]: any;
+    }>;
+    params: any;
+  }) => Promise<{
+    stream: any;
+    response?: any;
+    [key: string]: any;
+  }>;
+}
+
+/**
+ * Configuration options for the AI SDK middleware
+ */
+export interface MiddlewareConfig {
+  /** Enable debug logging */
+  debug?: boolean;
+  /** Name identifier for the middleware instance */
+  name?: string;
+}
+
+function detectProviderFromResult(result: any): string | undefined {
+  if (!result?.providerMetadata) {
+    return undefined;
+  }
```

Comment:
(Minor, style)
```suggestion
  if (!result?.providerMetadata) return undefined;
```

Reply:
we don't have this well defined yet. going to skip for now, but we could lock this down (`curly`)